packages/form-js-viewer/src/render/components/form-fields/parts/SimpleSelect.js

```diff
@@ -32,9 +33,17 @@ export default function SimpleSelect(props) {
   const selectRef = useRef();
 
   const {
-    state: loadState,
-    values: options
-  } = useValuesAsync(field);
+    loadState,
+    options
+  } = useOptionsAsync(field);
+
+  useRestrictedSingleSelectValue({
+    field,
+    loadState,
+    options,
+    value,
+    onChange: props.onChange
+  });
```

Comment:
This looks a bit like black magic. Looking inside the hook reveals it's simply another effect, but maybe we can improve the readability a bit to indicate we might updating the values?

Maybe something like `useCleanupSingleSelectValue`, something that is signalizing an action.